src/components/CreateAnalysis/CreateAnalysisWrapper.tsx

```diff
@@ -7,19 +7,47 @@ import { DcmImage } from "../../context/reducers/dicomImagesReducer";
 import CreateAnalysisService from "../../services/CreateAnalysisService";
 import ConfirmAnalysis from './ConfirmAnalysis';
 import CreateAnalysisDetail from "./CreateAnalysisDetail";
+import pacs_integration from '../../services/pacs_integration';
+import chris_integration from '../../services/chris_integration';
 
 const CreateAnalysisWrapper = () => {
   const { state: { dcmImages, createAnalysis: { selectedStudyUIDs } }, dispatch } = useContext(AppContext)
   const [isExpanded, setIsExpanded] = useState(false);
   const [isModalOpen, setIsModalOpen] = useState(false);
   const history = useHistory();
 
-  const submitAnalysis = () => {
-    const imagesSelected: DcmImage[] = CreateAnalysisService.pickImages(dcmImages, selectedStudyUIDs);
+  const submitAnalysis = async () => {
+    let imagesSelected: DcmImage[] = CreateAnalysisService.pickImages(dcmImages.filteredDcmImages, selectedStudyUIDs);
```

Comment:
From performance POV, should we use `pickImages()` that iterates over all images and filters out the ones that are not selected?
OR
should we have a dedicated array that keeps the copy of the selected images
OR
maybe its better to keep images in store as a dictionary where `keys` are `id` and we could keep track of selected `id`s.

^ this is something to consider when weighing pros & cons.

This can be further discussed once we know how PACS responses actually look like and their sizes. As an example, if the list of images are going to be short, iterating during filtering is not a big deal at all.

Reply:
Good points, will make a note of this.